source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -179,6 +179,22 @@ Http::FilterHeadersStatus Filter::encodeHeaders(Http::ResponseHeaderMap& headers
     // We can only insert trailers at the end of data, so keep track of this value
     // until then.
     grpc_status_ = grpcStatusFromHeaders(headers);
+
+    // This is a header-only response, and we should prepend the gRPC frame
+    // header directly.
+    if (end_stream && withhold_grpc_frames_) {
+      Envoy::Buffer::OwnedImpl data;
+      buildGrpcFrameHeader(data, 0);
+      encoder_callbacks_->addEncodedData(data, false);
+
+      // This call exists to ensure that the content-length is set correctly,
+      // regardless of whether we are withholding grpc frames above.
+      headers.setContentLength(Grpc::GRPC_FRAME_HEADER_SIZE);
+
+      // Insert grpc-status trailers to communicate the error code.
+      auto& trailers = encoder_callbacks_->addEncodedTrailers();
+      trailers.setGrpcStatus(grpc_status_);
```

Comment:
as a funcional change to the daa plane (even a bug fix) this should be runtime guarded and have a release note, thanks!

Reply:
@alyssawilk The [runtime guarding documentation](https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md#runtime-guarding) states the following:

> Runtime guarded features may either set true (running the new code by default) in the initial PR, after a testing interval, or during the next release cycle, at the PR author's and reviewing maintainer's discretion.

Do you have a preference for whether this is set to true by default?